### PR TITLE
[ML] fixing datafeed preview after allowing datafeed_config in job_config

### PR DESCRIPTION
**Pull request by benwtrent**
Since PR: https://github.com/elastic/elasticsearch/pull/74265

It is valid to supply a `datafeed_config` inside of a job config. But, the `_preview` API allows BOTH
a `datafeed_config` and `job_config` to be top-level objects. This can cause confusion and allowing
both a top level `datafeed_config` and an internal `job_config.datafeed_config` is unsupported 
behavior.

This commit protects against these weird scenarios and also strictly allows only a `job_config` with
a nested `datafeed_config` to be provided.

Consequently, on preview, the valid combinations are:

- Datafeed config ID only
- top level datafeed_config and top level job_config (without a nested datafeed_config)
- top level job_config with a nested datafeed_config

everything else is considered invalid

### CODE REVIEW

**Comment by benwtrent** [comment]
Flagged as `non-issue` as I think this should have originally been done in https://github.com/elastic/elasticsearch/pull/74265 but was an oversight.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedAction.java [hunk 1]
```diff
@@ -178,8 +178,19 @@ public Request build() {
...
-                    if (datafeedBuilder == null) {
-                        throw new IllegalArgumentException("[datafeed_config] must be present when a [job_config] is provided");
+                    if (datafeedBuilder == null && jobBuilder.getDatafeedConfig() == null) {
+                        throw new IllegalArgumentException(
+                            "[datafeed_config] must be present when a [job_config.datafeed_config] is not present"
+                        );
+                    }
+                    if (datafeedBuilder != null && jobBuilder.getDatafeedConfig() != null) {
+                        throw new IllegalArgumentException(
+                            "[datafeed_config] must not be present when a [job_config.datafeed_config] is provided"
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedAction.java:188]: nit: How about using the word `present` consistently in both messages. So: provided -> present

**Review by przemekwitek — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedAction.java (hunk 1 above)
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedAction.java:188]:
```suggestion
                            "[datafeed_config] must not be present when a [job_config.datafeed_config] is present"
```

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedActionRequestTests.java [hunk 2]
```diff
@@ -62,6 +76,25 @@ public void testValidation() {
...
+        requestBuilder
+            .setJobBuilder(
+                JobTests.buildJobBuilder(jobId)
+                    .setDatafeed(new DatafeedConfig.Builder(DatafeedConfigTests.createRandomizedDatafeedConfig(jobId)))
+            )
+            .setDatafeedId(null)
+            .setDatafeedBuilder(new DatafeedConfig.Builder());
+        ex = expectThrows(IllegalArgumentException.class, requestBuilder::build);
+        assertThat(ex.getMessage(),
+            containsString("[datafeed_config] must not be present when a [job_config.datafeed_config] is provided"));
```
Comment (benwtrent) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PreviewDatafeedActionRequestTests.java:89]:
```suggestion
            containsString("[datafeed_config] must not be present when a [job_config.datafeed_config] is present"));
```

**Pull request merged**